docs/dynamic_configuration.rst

```diff
@@ -49,19 +62,162 @@ Parameters would be applied in the following order (run-time are given the highe
+This allows configuration for all the nodes (2), configuration for a specific node using `ALTER SYSTEM` (3) and ensures that parameters essential to the running of Patroni are enforced. (4)
 
 
-Also, the following patroni configuration options can be changed dynamically:
+Also, the following patroni configuration options can be changed only dynamically:
 
-- ttl
-- loop_wait
-- retry_timeouts (to be defined first in patroni.yaml)
+- ttl: 30
+- loop_wait: 10
+- retry_timeouts: 10
+- maximum_lag_on_failover: 1048576
+- postgresql.use_slots: true
 
 Upon changing those options, Patroni should read the relevant section of the configuration stored in DCS and change their
 run-time values.
 
-Patroni nodes should dump the state of the DCS options to disk on startup and upon every change of the configuration.
-Only master is allowed to restore those options from the on-disk dump if those are completely absent from the DCS or invalid.
-
+Patroni nodes are dumping the state of the DCS options to disk upon every change of the configuration into file ``patroni.dynamic.json`` located in the postgres data directory. Only master is allowed to restore those options from the on-disk dump if those are completely absent from the DCS or invalid.
+
+REST API
+========
+
+We are providing REST API endpoint for working with dynamic configuration.
+
+GET /config
+-----------
+Get current version of dynamic configuration.
+
+.. code-block:: bash
+
+	$ curl -s localhost:8008/config | jq .
+	{
+	  "ttl": 30,
+	  "loop_wait": 10,
+	  "retry_timeout": 10,
+	  "maximum_lag_on_failover": 1048576,
+	  "postgresql": {
+	    "use_slots": true,
+	    "use_pg_rewind": true,
+	    "parameters": {
+	      "hot_standby": "on",
+	      "wal_log_hints": "on",
+	      "wal_keep_segments": 8,
+	      "wal_level": "hot_standby",
+	      "max_wal_senders": 5,
+	      "max_replication_slots": 5,
+	      "max_connections": "100"
+	    }
+	  }
+	}
+
+PATCH /config
+-------------
+Patch existing configuration.
+
+.. code-block:: bash
+
+	$ curl -s -XPATCH -d \
+		'{"loop_wait":5,"ttl":20,"postgresql":{"parameters":{"max_connections":"101"}}}' \
+		http://localhost:8008/config | jq .
+	{
+	  "ttl": 20,
+	  "loop_wait": 5,
+	  "maximum_lag_on_failover": 1048576,
+	  "retry_timeout": 10,
+	  "postgresql": {
+	    "use_slots": true,
+	    "use_pg_rewind": true,
+	    "parameters": {
+	      "hot_standby": "on",
+	      "wal_log_hints": "on",
+	      "wal_keep_segments": 8,
+	      "wal_level": "hot_standby",
+	      "max_wal_senders": 5,
+	      "max_replication_slots": 5,
+	      "max_connections": "101"
+	    }
+	  }
+	}
+
+Above REST API call patches existing configuration and returns the new configuration.
+
+Let's check that node processed this configuration. First of all it should start printing logs lines every 5 seconds (loop_wait=5). Change of "max_connections" requires restart, so "restart_pending" flag should be exposed:
```

Comment:
s/logs lines/log lines. 
..Changing the value of the "max_connections" requires restart...